src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -5,14 +5,1181 @@
+                _titleChangeHandlers(GetTitle());
+            }
+            CATCH_LOG();
+        }
+    }
+
+    // Method Description:
+    // - Called when our tab content size changes. This updates each tab with
+    //   the new size, so they have a chance to update each of their panes with
+    //   the new size.
+    // Arguments:
+    // - e: the SizeChangedEventArgs with the new size of the tab content area.
+    // Return Value:
+    // - <none>
+    void TerminalPage::_OnContentSizeChanged(const IInspectable& /*sender*/, Windows::UI::Xaml::SizeChangedEventArgs const& e)
+    {
+        const auto newSize = e.NewSize();
+        for (auto& tab : _tabs)
+        {
+            tab->ResizeContent(newSize);
+        }
+    }
+
+    // Method Description:
+    // - Called when the settings button is clicked. Launches a background
+    //   thread to open the settings file in the default JSON editor.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TerminalPage::_SettingsButtonOnClick(const IInspectable&,
+                                              const RoutedEventArgs&)
+    {
+        LaunchSettings();
+    }
+
+    // Method Description:
+    // - Responds to the TabView control's Tab Closing event by removing
+    //      the indicated tab from the set and focusing another one.
+    //      The event is cancelled so App maintains control over the
+    //      items in the tabview.
+    // Arguments:
+    // - sender: the control that originated this event
+    // - eventArgs: the event's constituent arguments
+    void TerminalPage::_OnTabClosing(const IInspectable& sender, const MUX::Controls::TabViewTabClosingEventArgs& eventArgs)
+    {
+        const auto tabViewItem = eventArgs.Item();
+        _RemoveTabViewItem(tabViewItem);
+
+        // If we don't cancel the event, the TabView will remove the item itself.
+        eventArgs.Cancel(true);
+    }
+
+    // Method Description:
+    // - Called when the feedback button is clicked. Launches github in your
+    //   default browser, navigated to the "issues" page of the Terminal repo.
+    void TerminalPage::_FeedbackButtonOnClick(const IInspectable&,
+                                              const RoutedEventArgs&)
+    {
+        const auto feedbackUriValue = Windows::ApplicationModel::Resources::ResourceLoader::GetForCurrentView().GetString(L"FeedbackUriValue");
+
+        winrt::Windows::System::Launcher::LaunchUriAsync({ feedbackUriValue });
+    }
+
+    // Method Description:
+    // - Called when the about button is clicked. See _ShowAboutDialog for more info.
+    // Arguments:
+    // - <unused>
+    // Return Value:
+    // - <none>
+    void TerminalPage::_AboutButtonOnClick(const IInspectable&,
+                                           const RoutedEventArgs&)
+    {
+        _ShowAboutDialog();
+    }
+
+    TerminalApp::TabRowControl* TerminalPage::GetTabRow()
+    {
+        return &_tabRow;
```

Comment:
>& [](start = 15, length = 1)

Why?  Can't the return type be TerminalApp::TabRowControl without the '*', eliminating the address-of here and the deref in the caller?

Reply:
Changed to return the object itself